src/qililab/calibration/calibration_node.py

```diff
@@ -758,19 +594,16 @@ def _json_serialize(_object: Any):
     Args:
         _object (Any): Object to serialize
     """
-    if isinstance(_object, dict):
-        for k, v in _object.items():
-            _object[k] = _json_serialize(v)
+    if isinstance(_object, np.ndarray):
+        return _json_serialize(_object.tolist())
 
-    if isinstance(_object, list):
-        for idx, elem in enumerate(_object):
-            _object[idx] = _json_serialize(elem)
+    if isinstance(_object, (list, tuple)):
+        return [_json_serialize(elem) for elem in _object]
 
-    if isinstance(_object, tuple):
-        tuple_list = [_json_serialize(elem) for elem in _object]
-        return tuple(tuple_list)
+    if isinstance(_object, dict):
+        return {_json_serialize(k): _json_serialize(v) for k, v in _object.items()}
```

Comment:
Nice improvement 💯

Reply:
Thanks!

(Also just to mention, the tuple now gets converted into list directly in this function, instead of doing a tuple, and then the `json.dump`,having to convert it himself :) )